app/src/main/java/com/hitchpeak/keystone/appServices/LocationShareService.kt

```diff
@@ -0,0 +1,78 @@
+package com.hitchpeak.keystone.appServices
+
+import android.Manifest
+import android.app.Notification
+import android.app.Service
+import android.content.Intent
+import android.content.pm.PackageManager
+import android.os.IBinder
+import android.support.v4.app.NotificationCompat
+import android.support.v4.content.ContextCompat
+import com.google.android.gms.location.*
+import com.hitchpeak.keystone.BuildConfig
+import com.hitchpeak.keystone.R
+import com.hitchpeak.keystone.models.LocationShareModel
+import com.hitchpeak.keystone.utils.HttpClient
+
+class LocationShareService : Service() {
+
+    protected lateinit var locationProvider: FusedLocationProviderClient
+
+    val channelId = BuildConfig.NOTIF_CHANNEL_ID
+    val notificationId = BuildConfig.LOCATION_NOTIF_ID
+
+    override fun onStartCommand(intent: Intent?, flags: Int, startId: Int): Int {
+        startLocationSharing()
+        startForegroundNotification()
+
+        return super.onStartCommand(intent, flags, startId)
+    }
+
+    private fun startForegroundNotification() {
+        val notification: Notification = NotificationCompat.Builder(this, channelId)
+                .setSmallIcon(R.drawable.ic_launcher_background)
+                .setContentTitle(getString(R.string.app_title))
+                .setContentText(getString(R.string.notification_text))
+                .setPriority(NotificationCompat.PRIORITY_HIGH)
+                .build()
+
+        startForeground(notificationId, notification)
+    }
+
+    private fun startLocationSharing() {
+        // The gist of location sharing:
+        locationProvider = LocationServices.getFusedLocationProviderClient(applicationContext)
+
+        val locationRequest = LocationRequest.create()
+        locationRequest.priority = LocationRequest.PRIORITY_HIGH_ACCURACY
```

Comment:
I think that value of this variable should also be in BuildConfigs, if we want to be consistent

Reply:
Yeah. I've added a switch so that the value is read from config now but the actual assignment is still from LocationRequest class in case it changes sometime in future.